Auto-open configured group windows at startup

jrick opened the pull request:
A [groups] section is added to the zkclient config file and must
specify 'index = name' pairs for each group name and the window index.

This is handy to keep groupchat windows in a common location across
client restarts.

marcopeereboom commented on zkclient/zkclient.conf.go:
```diff
@@ -44,5 +44,12 @@ profiler = 127.0.0.1:6061
+# Joined groups can be auto-opened in a specified order.
+# Each key describes the window index, and the value must be the group name.
+# Indexes begin at 1 due to index 0 being reserved for the console window.
+[groups]
```
Can we get rid of the section and use the following notation instead:
`group[1] = firstgroup`
`group[2] = secondgroup`

jrick replied:
this would complicate the parsing as well as the error handling when indexes are missing, but could be done.

because the ini package decodes into a map, we can't (easily) iterate through the settings in the file order either. using a unique section for only these settings allows us to at least initially know how many k/v fields were set.

I'm also thinking that we should use autoopen as the section name (or top-level 'autoopen[N]' keys) in case we want to distinguish between groups and direct chats in the value.

marcopeereboom replied:
Sections in INI are worse than INI itself imo. Really would prefer named arrays of sorts. `chat[1]` `group[3]` etc.

jrick committed:
Auto-open configured group windows at startup

A [groups] section is added to the zkclient config file and must
specify 'index = name' pairs for each group name and the window index.

This is handy to keep groupchat windows in a common location across
client restarts.

jrick committed:
error handling
